pkg/kotsadm/kotsadm.go

```diff
@@ -427,13 +427,16 @@ func updateKotsadmService(existing, desiredService *corev1.Service) *corev1.Serv
 
 // isKotsadmClusterScoped determines if the kotsadm pod should be running
 // with cluster-wide permissions or not
-func isKotsadmClusterScoped(applicationMetadata []byte) (bool, error) {
-	if len(applicationMetadata) == 0 {
+func isKotsadmClusterScoped(deployOptions *types.DeployOptions) (bool, error) {
+	if len(deployOptions.ApplicationMetadata) == 0 {
+		if deployOptions.UseMinimalRBAC {
```

Comment:
doesn't this mean that if the app _doesn't_ support minimal rbac (because it has no app spec), the user can still opt-in? that feels wrong to me